modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -1089,3 +1091,22 @@ def test_setitem_bool_preserve_dtypes():
     # scalar as a col_loc
     df.loc[indexer, "a"] = 2.0
     assert df._query_compiler._modin_frame.has_materialized_dtypes
+
+
+@pytest.mark.parametrize(
+    "modify_config", [{ExperimentalGroupbyImpl: True}], indirect=True
+)
+def test_groupby_size_shuffling(modify_config):
+    # verifies that 'groupby.size()' works with reshuffling implementation
+    # https://github.com/modin-project/modin/issues/6367
+    df = pd.DataFrame({"a": [1, 1, 2, 2], "b": [3, 4, 5, 6]})
+    modin_frame = df._query_compiler._modin_frame
+
+    with mock.patch.object(
+        modin_frame,
+        "_apply_func_to_range_partitioning",
+        wraps=modin_frame._apply_func_to_range_partitioning,
+    ) as shuffling_method:
+        try_cast_to_pandas(df.groupby("a").size())
+
+    shuffling_method.assert_called()
```

Comment:
what assert are we expecting there?

Reply:
The test verifies that setting the `ExperimentalGroupbyImpl` to True will guarantee us to engage the reshuffling groupby implementation. 

[The patched method](https://github.com/modin-project/modin/blob/947e06bb07ae93633e2b0a68ea9d320f689d5274/modin/core/dataframe/pandas/dataframe/dataframe.py#L2289-L2310) is the one that actually builds range-partitioning and executes functions over it, so if the code made it through to this method, we can assume that the reshuffling implementation was used, rather [than the default MapReduce one.](https://github.com/modin-project/modin/blob/947e06bb07ae93633e2b0a68ea9d320f689d5274/modin/core/storage_formats/pandas/query_compiler.py#L3267-L3271)